spec/apps/gcm_app_spec.rb

```diff
@@ -0,0 +1,69 @@
+module RailsPushNotifications
+  describe GCMApp, type: :model do
+
+    it 'creates an instance given valid attributes' do
+      GCMApp.create! attributes_for :gcm_app
+    end
+
+    describe 'validations' do
+      let(:app) { build :gcm_app }
+
+      it 'requires a gcm key' do
+        app.gcm_key = nil
+        expect(app).to_not be_valid
+      end
+    end
+
+    describe 'notifications relationship' do
+      let(:app) { create :gcm_app }
+
+      it 'can create new notifications' do
+        expect do
+          app.notifications.create attributes_for :gcm_notification
+        end.to change { app.reload.notifications.count }.from(0).to 1
+      end
+    end
+
+    describe '#push_notifications' do
+      let(:app) { create :gcm_app }
+      let(:notifications) do
+        (1..10).map { create :gcm_notification, app: app }
+      end
+      let(:single_notification) { create :gcm_notification, app: app }
+
+      let(:response) { JSON.dump a: 1 }
+
+      before do
+        stub_request(:post, 'https://android.googleapis.com/gcm/send').
+          to_return status: [200, 'OK'], body: response
+      end
+
+      it 'assigns results' do
+        expect do
+          app.push_notifications
+        end.to change {
+          notifications.map { |n| n.reload.results }
+          }.from([nil] * 10).
```

Comment:
`end` at 46, 10 is not aligned with `change {` at 44, 15 or `end.to change {` at 44, 8